OpenRA.Game/Primitives/PriorityQueue.cs

```diff
@@ -41,10 +43,11 @@ public void Add(T item)
 		{
 			var addLevel = level;
 			var addIndex = index;
+			T above;
 
-			while (addLevel >= 1 && comparer.Compare(Above(addLevel, addIndex), item) > 0)
+			while (addLevel >= 1 && comparer.Compare((above = Above(addLevel, addIndex)), item) > 0)
```

Comment:
We generally try to avoid using inline assignments (not sure the correct term) like this. Can we restructure the loop so that above is assigned on its own line?